dkg/keycast.go

```diff
@@ -121,7 +122,11 @@ func leadKeyCast(ctx context.Context, tp kcTransport, def cluster.Definition) ([
 
 	log.Info(ctx, "Node selected as dealer, serving shares...")
 
+	var serveMutex sync.Mutex
```

Comment:
suggest defining this in variable group above and add comment, `shared server state`